File: Python/String-2/cat_dog.py
"""
Dada uma String "str", retorne true se nela possuir o mesmo número de ocorrências
das strings "cat" e "dog".
Ex.:(('catdog') → True; ('1cat1cadodog') → True; ('catcat') → False).
"""


def cat_dog(str):
    return str.count("cat") == str.count("dog")


print(cat_dog("catdog"))

File: Java/String-2/src/catDog/CatDog.java
package catDog;

public class CatDog {
    //Retorne true se as Strings "cat" e "dog" aparecem o mesmo número de vezes na String dada.

    public boolean catDog(String str) {
        int cat = 0, dog = 0;
        for (int i = 0; i < str.length()-2; i++){
            if (str.startsWith("cat", i)) cat ++;
            if (str.startsWith("dog", i)) dog ++;
        }
        return cat == dog;
    }
}
